jerry-core/ecma/operations/ecma-big-uint.c

```diff
@@ -0,0 +1,1252 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ecma-big-uint.h"
+#include "ecma-helpers.h"
+#include "jmem.h"
+#include "lit-char-helpers.h"
+
+#if ENABLED (JERRY_BUILTIN_BIGINT)
+
+JERRY_STATIC_ASSERT (sizeof (ecma_bigint_two_digits_t) == 2 * sizeof (ecma_bigint_digit_t),
+                     ecma_big_int_two_digits_must_be_twice_as_long_as_ecma_big_int_digit);
+
+JERRY_STATIC_ASSERT ((1 << ECMA_BIGINT_DIGIT_SHIFT) == (8 * sizeof (ecma_bigint_digit_t)),
+                     ecma_bigint_digit_shift_is_incorrect);
+
+/**
+ * Create a new BigInt value
+ *
+ * @return new BigInt value, NULL on error
+ */
+ecma_extended_primitive_t *
+ecma_bigint_create (uint32_t size) /**< size of the new BigInt value */
+{
+  JERRY_ASSERT ((size % sizeof (ecma_bigint_digit_t)) == 0);
+
+  if (JERRY_UNLIKELY (size > ECMA_BIGINT_MAX_SIZE))
+  {
+    return NULL;
+  }
+
+  ecma_extended_primitive_t *value_p;
+
+  if (size == 0)
+  {
+    value_p = (ecma_extended_primitive_t *) jmem_pools_alloc (sizeof (ecma_extended_primitive_t));
```

Comment:
Are zero sized bigints that common to have a special case for them?

Reply:
This code is copied from external API error handling. On the long term I have been thinking about making bigint zero a special constant.